cli/runner/cmd/requests.go

```diff
@@ -82,6 +86,28 @@ func postSourceFile(server string, endpoint string, filepath string, langCheck s
 	req.Header.Add("Content-Type", "application/json")
 
 	//TODO: implement CLI check language logic before adding cookie
+	if langCheck == "implicit" {
+		langs, err := getLangListJSON(server, LANG_ENDPOINT)
```

Comment:
Ah I see how this checking the supported languages first can complicate things too much. How about skip checking the endpoint and just do the filetype matching?